src/libcrun/cgroup-utils.c

```diff
@@ -286,7 +290,9 @@ rmdir_all_fd (int dfd)
           for (i = 0; i < n_pids; i++)
             kill (pids[i], SIGKILL);
 
-          return rmdir_all_fd (child_dfd);
+          tmp = child_dfd;
+          child_dfd = -1;
+          return rmdir_all_fd (tmp);
```

Comment:
nit just in case you are going to push a new version:

we could avoid the assignment to `tmp`, with `return rmdir_all_fd (get_and_reset (&child_dfd));`

Reply:
Already confirmed from @giuseppe this will be done later for entire codebase.